examples/stein_dmm.py

```diff
@@ -0,0 +1,379 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Deep Markov Model inferred using SteinVI
+=================================================
+An implementation of a Deep Markov Model in NumPyro based on reference [1][2] and the Pyro DMM example.
+This is essentially the DKS variant outlined in the paper.
```

Comment:
Could you sketch a bit more the purpose of this model and this dataset (it is not clear to me what is the purpose of this fancy model based on the current description) and add link to the Pyro example to the reference section? FYI you can add thumbnail images to  the examples (in case you think it is worth).

Reply:
I'll update the model description and develop an image of the generated music and some inference/model diagnostics.